src/js/components/productSupplier/create/subsections/PreferenceTypeVariations.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import { useFieldArray } from 'react-hook-form';
+import { RiAddLine } from 'react-icons/ri';
+
+import DataTable from 'components/DataTable';
+import Button from 'components/form-elements/Button';
+import Subsection from 'components/Layout/v2/Subsection';
+import InvalidItemsIndicator from 'components/productSupplier/create/InvalidItemsIndicator';
+import usePreferenceTypeVariationsColumns
+  from 'hooks/productSupplier/form/usePreferenceTypeVariationsColumns';
+
+const PreferenceTypeVariations = ({ control, errors: { productSupplierPreferences } }) => {
```

Comment:
I don't like the naming (productSuppllierPreferences). Let's stick to errors or at least pass it to the hook like:
```js
usePreferenceTypeVariationsColumns({
    errors: productSupplierPreferences,
    control,
    remove,
  });
```
and use it in the hook via `errors`.